File: src/main/java/com/polygloat/dtos/request/ResetPasswordRequest.java
package com.polygloat.dtos.request;

import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;

@Data
@NoArgsConstructor
public class ResetPasswordRequest {
    @NotBlank
    String callbackUrl;

    @Email
    String email;
}

File: src/main/java/io/polygloat/dtos/request/ResetPasswordRequest.kt
package io.polygloat.dtos.request

import org.springframework.validation.annotation.Validated
import javax.validation.constraints.Email
import javax.validation.constraints.NotBlank

@Validated
data class ResetPasswordRequest(
        @NotBlank
        var callbackUrl: String? = null,

        @Email
        var email: String? = null
)